include/jemalloc/internal/edata.h

```diff
@@ -288,10 +288,53 @@ edata_szind_get(const edata_t *edata) {
 }
 
 static inline size_t
-edata_usize_get(const edata_t *edata) {
+edata_usize_get_from_ind_unsafe(const edata_t *edata) {
 	return sz_index2size(edata_szind_get(edata));
 }
 
+static inline size_t
+edata_usize_get_from_ind(const edata_t *edata) {
```

Comment:
This doesn't appear to be used.

It looks the unsafe helper above is only used to implement `edata_usize_get`. In this case, instead provide an unsafe helper and we evaluate if the call site should be using it, we should simply avoid providing such helper. In other words, we should have no function returning an invalid usize to begin with.

To do this:

```
edata_usize_get() {
  szind = edata_szind_get(edata);
  if (szind <= ...) {
    return sz_index2size(edata_szind_get(edata));
  }
  // impl of edata_usize_get_from_size
}
```

In the end, we still only provide a single `edata_usize_get helper`, w/o any unsafe helper only for certain cases.